### PR TITLE
`amp-toolbox-cache-url`: Added Support for Fallback AMP Curl Domains & Package Maintenance

### PR DESCRIPTION
**Pull request by torch2424**
Hello,

This is a follow up to https://github.com/ampproject/amp-viewer/pull/74

This accomplishes the following things:
* Update README(s)
* moves the Amp CURL logic from viewer into `amp-toolbox-cache-url`, along with some changes to the API to expose getting both the cache URL, and the CURLs subdomain.
* replaces the `mime-types` package with file extension pathname matching, to drastically reduce the size of the package for use within a browser environment.
* Moves over tests from the amp-viewer, to confirm the Curls subdomains are working as expected.

This PR should be carefully reviewed, however, most of the code comes from the amp-viewer project, and has already been approved by them.

My apologies for the large-ish PR. I can definitely move this into multiple PRs if desired (one for the mime types, one for the tests, and one for the Curl Subdomain). A followup PR will made in addition to this one for general project infrastructure, using rollup.

File Extensions come from the following MIT Licensed github repos:

* https://github.com/arthurvr/image-extensions
* https://github.com/awcross/font-extensions

### CODE REVIEW

File: cache-url/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "amp-toolbox-cache-url",
-  "version": "0.1.0",
+  "version": "1.0.0",
```
Comment (sebastianbenz) [inline cache-url/package.json]: 👍

File: README.md [hunk 2]
```diff
@@ -23,6 +23,7 @@ A collection of AMP tools making it easier to publish and host AMP pages. The fo
 - **[amp-optimizer](./optimizer):** a javascript library implementing server-side-rendering for AMP pages.
 - **[amp-optimizer-express](./optimizer-express)** an [express](http://expressjs.com/) middleware that applies AMP server-side-rendering on the fly.
 - **[amp-runtime-version](./runtime-version):** a javascript library for querying the current AMP runtime version.
+- **[amp-cache-url](./cache-url):** a javascript library for translating origin URLs to the AMP Cache URL format.
```
Comment (sebastianbenz) [inline README.md]: Can you link AMP Cache URL format to https://developers.google.com/amp/cache/overview

File: cache-url/lib/AmpCacheUrlGenerator.js [hunk 3]
```diff
@@ -17,45 +17,49 @@
...
 const {URL} = require('url');
-const punycode = require('punycode');
-const mime = require('mime-types');
+const createCurlsSubdomain = require('./AmpCurlUrlGenerator');
 
 /**
  * Translates the canonicalUrl to the AMP Cache equivalent, for a given AMP Cache.
+ * Example:
+ * createCacheUrl('cdn.ampproject.org', 'https://hello-world.com')
+ * // Should resolve: 'https://hello--world-com.cdn.ampproject.org/c/s/hello-world.com'
```
Comment (sebastianbenz) [inline cache-url/lib/AmpCacheUrlGenerator.js]: Using `//` in here looks a bit weird, I'd suggest removing it.

**Comment by torch2424** [comment]
@sebastianbenz To get this working in browser it will require that #102 is pulled in as well 😄 

After both of these PRs get merged, I'll go ahead and do a "clean up PR" to do any additional testing, and prep before deploying to npm (Like hosting a headless chrome window in our travis builds and testing for both node and browser if you would like).

File: cache-url/lib/AmpCacheUrlGenerator.js [hunk 4]
```diff
@@ -17,45 +17,47 @@
...
-  cacheUrl.hostname = punycode.toASCII(unicodeHostname) + '.' + domainSuffix;
-  cacheUrl.pathname = pathSegment + originalHostname + cacheUrl.pathname;
-  return cacheUrl.toString();
+  return createCurlsSubdomain(canonicalUrl.toString()).then((curlsSubdomain) => {
+    const cacheUrl = new URL(url);
+    cacheUrl.protocol = 'https';
+    cacheUrl.hostname = curlsSubdomain + '.' + domainSuffix;
+    cacheUrl.pathname = pathSegment + canonicalUrl.hostname + canonicalUrl.pathname;
+    return cacheUrl.toString();
+  }).catch((err) => {
```
[the remark this answers is not in the document]
Comment (torch2424) [reply cache-url/lib/AmpCacheUrlGenerator.js]: Done

**Pull request merged**